[Console] Add icons to main panel

ElenaStoeva opened the pull request:
## Summary

This PR adds icon buttons to main panel (currently they don't do anything).



https://github.com/user-attachments/assets/b8223574-3a80-4241-b3c5-5017eb769a70




### Checklist

Delete any items that are not applicable to this PR.

- [ ] Any text added follows [EUI's writing guidelines](https://elastic.github.io/eui/#/guidelines/writing), uses sentence case text and includes [i18n support](https://github.com/elastic/kibana/blob/main/packages/kbn-i18n/README.md)
- [ ] [Documentation](https://www.elastic.co/guide/en/kibana/master/development-documentation.html) was added for features that require explanation or tutorials
- [ ] [Unit or functional tests](https://www.elastic.co/guide/en/kibana/master/development-tests.html) were updated or added to match the most common scenarios
- [ ] [Flaky Test Runner](https://ci-stats.kibana.dev/trigger_flaky_test_runner/1) was used on any tests changed
- [ ] Any UI touched in this PR is usable by keyboard only (learn more about [keyboard accessibility](https://webaim.org/techniques/keyboard/))
- [ ] Any UI touched in this PR does not create any new axe failures (run axe in browser: [FF](https://addons.mozilla.org/en-US/firefox/addon/axe-devtools/), [Chrome](https://chrome.google.com/webstore/detail/axe-web-accessibility-tes/lhdoppojpmngadmnindnejefpokejbdd?hl=en-US))
- [ ] If a plugin configuration key changed, check if it needs to be allowlisted in the cloud and added to the [docker list](https://github.com/elastic/kibana/blob/main/src/dev/build/tasks/os_packages/docker_generator/resources/base/bin/kibana-docker)
- [ ] This renders correctly on smaller devices using a responsive layout. (You can test this [in your browser](https://www.browserstack.com/guide/responsive-testing-on-local-server))
- [ ] This was checked for [cross-browser compatibility](https://www.elastic.co/support/matrix#matrix_browsers)


### Risk Matrix

Delete this section if it is not applicable to this PR.

Before closing this PR, invite QA, stakeholders, and other developers to identify risks that should be tested prior to the change/feature release.

When forming the risk matrix, consider some of the following examples and how they may potentially impact the change:

| Risk                      | Probability | Severity | Mitigation/Notes        |
|---------------------------|-------------|----------|-------------------------|
| Multiple Spaces&mdash;unexpected behavior in non-default Kibana Space. | Low | High | Integration tests will verify that all features are still supported in non-default Kibana Space and when user switches between spaces. |
| Multiple nodes&mdash;Elasticsearch polling might have race conditions when multiple Kibana nodes are polling for the same tasks. | High | Low | Tasks are idempotent, so executing them multiple times will not result in logical error, but will degrade performance. To test for this case we add plenty of unit tests around this logic and document manual testing procedure. |
| Code should gracefully handle cases when feature X or plugin Y are disabled. | Medium | High | Unit tests will verify that any feature flag or plugin combination still results in our service operational. |
| [See more potential risk examples](https://github.com/elastic/kibana/blob/main/RISK_MATRIX.mdx) |


### For maintainers

- [ ] This was checked for breaking API changes and was [labeled appropriately](https://www.elastic.co/guide/en/kibana/master/contributing.html#kibana-release-notes-process)

ElenaStoeva committed:
[Console] Add icons to main panel

elasticmachine commented:
Pinging @elastic/kibana-management (Team:Kibana Management)

ryankeairns commented on src/plugins/console/public/application/containers/main/main.tsx:
```diff
@@ -51,19 +60,47 @@ export function Main({ isEmbeddable = false }: MainProps) {
+                  <EuiButtonIcon iconType="save" onClick={() => {}} />
+                </EuiFlexItem>
+                <EuiFlexItem grow={false}>
+                  <EuiButtonIcon iconType="keyboard" color="primary" />
```
Primary is the default; no need to include

ryankeairns commented on src/plugins/console/public/application/containers/main/main.tsx:
```diff
@@ -51,19 +60,47 @@ export function Main({ isEmbeddable = false }: MainProps) {
+                  <EuiButtonIcon iconType="keyboard" color="primary" />
+                </EuiFlexItem>
+                <EuiFlexItem grow={false}>
+                  <EuiButtonIcon iconType="questionInCircle" color="primary" />
```
Primary is the default; no need to include

ryankeairns commented on src/plugins/console/public/application/containers/main/main.tsx:
```diff
@@ -51,19 +60,47 @@ export function Main({ isEmbeddable = false }: MainProps) {
+                iconType="editorCodeBlock"
+                size="xs"
+                color="text"
+                css={{ fontWeight: 'normal' }}
```
Is this necessary @julianrosado ?

julianrosado replied:
Nah, I think it's fine with the default. That is probably more my fault for not being clearer, and Elena trying to closely match my design. @ElenaStoeva totally good to use the default here.

ryankeairns replied:
Thanks for clarifying. To be extra clear, we're suggesting to remove line 97.

ryankeairns commented on src/plugins/console/public/styles/_app.scss:
```diff
@@ -19,7 +19,7 @@
 
 .consoleTabs {
-  padding: 0 $euiSizeS;
+  padding: 0 $euiSizeM 0 $euiSizeS;
```
Looking at the screenshot, the padding between these icon buttons looks excessive. Can we tighten this up? For reference, see the icon buttons in the top right of Kibana... this represents the more common spacing/pattern.

julianrosado replied:
Agree with Ryan, we can shrink the spacing between the button icons. I think the mock has space small (8px) between button elements.

ElenaStoeva replied:
Sure, I added `gutterSize: "s"` to the EuiFlexGroup for these icons and this decreased the spacing. The mentioned padding from this class is for the whole top bar, including the menu tabs.

ryankeairns replied:
Good point. If you're using flex with gutter size, then you may not need these padding styles any longer.

ElenaStoeva replied:
I think these padding styles are still needed as they are for the whole top bar, while the gutter size is for the icons flex group only. I did remove the unnecessary padding sizes though.

ryankeairns reviewed (changes requested):
Added a few comments of things to consider. Thanks

julianrosado commented:
Ryan beat me to the punch with the feedback as usual. Once his feedback(s) is/are addressed I'll be good too!

ElenaStoeva committed:
Address feedback

ElenaStoeva commented:
Thanks for the feedback @ryankeairns and @julianrosado! Added your suggestions with the latest changes and updated the screenshots. Let me know if you have any further suggestions.

sabarasaba commented on src/plugins/console/public/application/containers/main/main.tsx:
```diff
@@ -51,19 +60,41 @@ export function Main({ isEmbeddable = false }: MainProps) {
+                  />
+                </EuiFlexItem>
+                <EuiFlexItem grow={false}>
+                  <EuiButtonIcon iconType="save" onClick={() => {}} />
```
I think we should add aria labels here otherwise will be quite hard to understand what these are with a screen reader. Also I was thinking perhaps we could add a little tooltip with a description of the action? Otherwise might be hard to understand for the user what these buttons do 🤔 cc: @julianrosado

ElenaStoeva replied:
Good call about the aria labels! Added them with [d9afe73](https://github.com/elastic/kibana/pull/190820/commits/d9afe73f98ba8c383fdd5abb460f8c882d08902f).

Yes, I think adding tooltips is a great idea! Let's see if @julianrosado has any thoughts on this.

julianrosado replied:
That's a _great_ idea @sabarasaba @ElenaStoeva ! I'm mad at myself for not including that to begin with... I actually went back to my prototype to check that 😫 Yes please feel free to add the tooltip! And there is never a situation where an additional aria label wouldn't be welcome. You'd be making our A11y team very happy... and it would make the console a good example of our a11y focus.

ElenaStoeva replied:
Added the tooltip and updated the screenshot in the PR!

sabarasaba reviewed (commented):
Code changes lgtm, left a small nit and an open question that I would like to clarify before we merge this in

ElenaStoeva committed:
Add aria labels to icon buttons

ElenaStoeva committed:
Remove unnecessary padding sizes

ryankeairns reviewed (approved):
Looks good! Thanks for the changes.

ElenaStoeva committed:
Add tooltips to icon buttons

ElenaStoeva committed:
Merge branch 'console-phase-2' into console-main-panel-icons

ElenaStoeva committed:
Remove redundant classname

ElenaStoeva committed:
Rename i18n const name

ElenaStoeva commented on src/plugins/console/public/styles/_app.scss:
```diff
@@ -41,11 +41,6 @@
   }
 }
 
-.conApp__outputPanel__emptyState {
```
Removing this as it was added by one of the first commits from https://github.com/elastic/kibana/pull/190627 but it's not needed anymore as we use an EUI empty prompt.

sabarasaba reviewed (approved):
latest lgtm!
